devicetypes/smartthings/aeon-led-bulb-6.src/aeon-led-bulb-6.groovy

```diff
@@ -0,0 +1,337 @@
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def linkText = device.label ?: device.name
+	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
+}
+
+def buildOffOnEvent(cmd){
+	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
+}
+
+def on() {
+	commands(buildOffOnEvent(0xFF), 5000)
+}
+
+def off() {
+	commands(buildOffOnEvent(0x00), 5000)
+}
+
+def refresh() {
+	commands([zwave.switchMultilevelV3.switchMultilevelGet()] + queryAllColors())
+}
+
+def ping() {
+	log.debug "ping().."
+	unschedule(offlinePing)
+	runEvery5Minutes(offlinePing)
+	command(zwave.switchMultilevelV3.switchMultilevelGet())
+}
+
+def offlinePing() {
+	log.debug "offlinePing()..."
+	sendHubCommand(new physicalgraph.device.HubAction(command(zwave.switchMultilevelV3.switchMultilevelGet())))
+}
+
+def setLevel(level) {
+	setLevel(level, 1)
+}
+
+def setLevel(level, duration) {
+	log.debug "setLevel($level, $duration)"
+	if(level > 99) level = 99
+	commands([
+		zwave.switchMultilevelV3.switchMultilevelSet(value: level, dimmingDuration: duration),
+		zwave.switchMultilevelV3.switchMultilevelGet(),
+	], (duration && duration < 12) ? (duration * 1000) : 3500)
+}
+
+def setSaturation(percent) {
+	log.debug "setSaturation($percent)"
+	setColor(saturation: percent)
+}
+
+def setHue(value) {
+	log.debug "setHue($value)"
+	setColor(hue: value)
+}
+
+def setColor(value) {
+	log.debug "setColor($value)"
+	def result = []
+	if (value.hex) {
+		def c = value.hex.findAll(/[0-9a-fA-F]{2}/).collect { Integer.parseInt(it, 16) }
+		result << zwave.switchColorV3.switchColorSet(red:c[0], green:c[1], blue:c[2], warmWhite:0, coldWhite:0)
+	} else {
+		def hue = value.hue ?: device.currentValue("hue")
+		def saturation = value.saturation ?: device.currentValue("saturation")
+		if(hue == null) hue = 13
+		if(saturation == null) saturation = 13
+		def rgb = huesatToRGB(hue, saturation)
+		result << zwave.switchColorV3.switchColorSet(red: rgb[0], green: rgb[1], blue: rgb[2], warmWhite:0, coldWhite:0)
+	}
+	commands(result) + "delay 7000" + commands(queryAllColors(), 1000)
+}
+
+private getCOLOR_TEMP_MAX() { 6500 }
+private getCOLOR_TEMP_MIN() { 2700 }
+private getCOLOR_TEMP_DIFF() { COLOR_TEMP_MAX - COLOR_TEMP_MIN }
+private getWARM_WHITE_CONFIG() { 0x51 }
+private getCOLD_WHITE_CONFIG() { 0x52 }
+
+def setColorTemperature(temp) {
+	if(temp > COLOR_TEMP_MAX)
+		temp = COLOR_TEMP_MAX
+	else if(temp < COLOR_TEMP_MIN)
+		temp = COLOR_TEMP_MIN
+	log.debug "setColorTemperature($temp)"
+	def warmValue = temp < 5000 ? 255 : 0
+	def coldValue = temp >= 5000 ? 255 : 0
+	def parameterNumber = temp < 5000 ? WARM_WHITE_CONFIG : COLD_WHITE_CONFIG
+	def cmds = [zwave.configurationV1.configurationSet([parameterNumber: parameterNumber, size: 2, scaledConfigurationValue: temp]),
+			zwave.switchColorV3.switchColorSet(red: 0, green: 0, blue: 0, warmWhite: warmValue, coldWhite: coldValue)]
```

Comment:
This is what you meant by having to use Configuration, right?

Basically if you want to control warm white you have to turn it on and vice versa for cold white?

What happens if you don't?

Reply:
So, this bulb has a "priority" attached to each of the LED color values. Warm white has the highest priority, then cold white, then RGB. If warm white has any value other than 0, it overrides everything else and only the warm LEDs will turn on.